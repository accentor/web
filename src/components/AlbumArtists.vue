<template>
  <div>
    <span v-for="aa of album.album_artists" :key="`${aa.artist_id} ${aa.name}`"
      ><RouterLink :to="{ name: 'artist', params: { id: aa.artist_id } }">{{
        aa.name
      }}</RouterLink
      >{{ aa.separator }}</span
    >
  </div>
</template>
<script>
export default {
  name: "AlbumArtists",
  props: {
    album: {}
  }
};
</script>
<style scoped>
a {
  text-decoration: none;
}
</style>
