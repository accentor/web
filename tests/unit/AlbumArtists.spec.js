import Vuetify from "vuetify";
import VueRouter from "vue-router";
import { mount, createLocalVue } from "@vue/test-utils";
import AlbumArtists from "@/components/AlbumArtists.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);

const factory = (values = {}, album_values = {}) => {
  const vuetify = new Vuetify();
  const router = new VueRouter();

  return mount(AlbumArtists, {
    localVue,
    vuetify,
    router,
    propsData: {
      album: {
        id: 1,
        title: "title",
        album_artists: [
          {
            artist_id: 2,
            name: "artist 2",
            separator: null,
            order: 2,
          },
          {
            artist_id: 1,
            name: "artist 1",
            separator: " / ",
            order: 1,
          },
        ],
        album_labels: [],
        edition: null,
        edition_description: null,
        release: "1970-01-01",
        ...album_values,
      },
      ...values,
    },
    mocks: {
      $t: (msg) => msg,
    },
  });
};

describe("AlbumArtists", () => {
  const wrapper = factory();

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders link for every artist", () => {
    expect(wrapper.findAll("a").length).toBe(2);
  });

  it("renders artists' names in order", () => {
    const routerLinkArray = wrapper.findAll("a");
    expect(routerLinkArray.at(0).text()).toMatch("artist 1");
    expect(routerLinkArray.at(1).text()).toMatch("artist 2");
  });

  it("has title with artists name", () => {
    expect(wrapper.attributes().title).toBe("artist 1 / artist 2");
  });
});

describe("AlbumArtists without artists", () => {
  const wrapper = factory({}, { album_artists: [] });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders link for every artist", () => {
    expect(wrapper.findAll("routerlink-stub").length).toBe(0);
  });
});
