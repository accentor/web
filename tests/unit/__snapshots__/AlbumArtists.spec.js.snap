// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumArtists renders correctly 1`] = `
<div
  class=""
  title="artist 1 / artist 2"
>
  <span>
    <a
      aria-current="page"
      class="router-link-exact-active router-link-active"
      href="#/"
    >
      artist 1
    </a>
     / 
  </span>
  <span>
    <a
      aria-current="page"
      class="router-link-exact-active router-link-active"
      href="#/"
    >
      artist 2
    </a>
    
  </span>
</div>
`;

exports[`AlbumArtists without artists renders correctly 1`] = `
<div>
  <span>
    music.artist.various
  </span>
</div>
`;
